### PR TITLE
Add docs for using transactionCreate and transactionUpdate mutations

### PR DESCRIPTION
**Pull request by korycins**
It extends a chapter related to finalizing the checkout by using new checkout flow

### CODE REVIEW

File: docs/developer/checkout.mdx [hunk 1]
```diff
@@ -1082,35 +1128,32 @@ This feature was introduced in **Saleor 3.2**.
 
 :::caution
 
-This feature is currently in **Feature Preview**. That means this part of Saleor is not complete
-but is available to experiment with and provide feedback, and is subject to change.
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
```
Comment (krzysztofwolski) [inline docs/developer/checkout.mdx:1131]: Seems there are unintended changes

File: docs/developer/checkout.mdx [hunk 2]
```diff
@@ -1147,9 +1190,146 @@ As a result, we get an order object:
...
+
+:::
+
+Transaction stores details of payment transaction attached to order or checkout.
+
+
+The [transactionCreate](developer/api-reference/mutations/transaction-create.mdx) mutation takes the following arguments:
+- `id`: The ID of the checkout or order.
+- `transaction`: Input data required to create a new transaction object.
+- `transactionEvent`: Data that defines a transaction event.
```
Comment (krzysztofwolski) [inline docs/developer/checkout.mdx:1213]: It was hard for me to understand what the Event for, before reading the example.

File: docs/developer/checkout.mdx [hunk 3]
```diff
@@ -1147,9 +1190,146 @@ As a result, we get an order object:
...
+The [transactionUpdate](developer/api-reference/mutations/transaction-update.mdx) mutation allows to update a details of transaction.
+It takes the following arguments:
+- `id`: The ID of the transaction.
+- `transaction`: Input data that will be used to update transaction object.
+- `transactionEvent`: Data that defines a transaction event.
+
+
+The following example shows how the [transactionUpdate](developer/api-reference/mutations/transaction-create.mdx) mutation 
+is used to update the transaction.
+The updated status of the transaction is `Captured`. The available action is REFUND.
```
Comment (krzysztofwolski) [inline docs/developer/checkout.mdx:1285]:
```suggestion
The updated status of the transaction is `CAPTURED`. The available action is `REFUND`. 
```

File: docs/developer/checkout.mdx [hunk 4]
```diff
@@ -1147,9 +1190,146 @@ As a result, we get an order object:
...
+It takes the following arguments:
+- `id`: The ID of the transaction.
+- `transaction`: Input data that will be used to update transaction object.
+- `transactionEvent`: Data that defines a transaction event.
+
+
+The following example shows how the [transactionUpdate](developer/api-reference/mutations/transaction-create.mdx) mutation 
+is used to update the transaction.
+The updated status of the transaction is `Captured`. The available action is REFUND. 
+The authorized funds are captured, so amountAuthorized is 0. amountCaptured is 99USD
```
Comment (krzysztofwolski) [inline docs/developer/checkout.mdx:1286]:
```suggestion
The authorized funds are captured, so `amountAuthorized` is 0 and `amountCaptured` is equal 99USD.
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

**Review by krzysztofwolski — APPROVED** [review]

File: docs/developer/checkout.mdx [hunk 5]
```diff
@@ -767,8 +767,9 @@ As a result, we get an updated checkout object with new checkout email set:
   }
 }
 ```
-
-## Payment
+## Finalizing checkout with Saleor's payment gateways
+The payment flow that uses a Saleor's payment gateway or payment synchronous webhooks to process a payment.
+### Payment
```
Comment (timuric) [inline docs/developer/checkout.mdx:774]: We should avoid mentioning JS, payments can be processed in other platforms than web

File: docs/developer/checkout.mdx [hunk 6]
```diff
@@ -1071,8 +1072,53 @@ A response would look like this:
 }
 ```
 
+## Finalizing checkout with Saleor App
+:::caution
+
+This feature is currently in **Feature Preview**. This part of Saleor is not complete
```
Comment (timuric) [inline docs/developer/checkout.mdx]: It is subject to change, but it is probably complete?

**Review by timuric — APPROVED** [review]

File: docs/developer/checkout.mdx [hunk 7]
```diff
@@ -767,8 +767,9 @@ As a result, we get an updated checkout object with new checkout email set:
   }
 }
 ```
-
-## Payment
+## Finalizing checkout with Saleor's payment gateways
+The payment flow that uses a Saleor's payment gateway or payment synchronous webhooks to process a payment.
+### Payment
 
 Depending on the selected payment gateway, you will either use the JavaScript form which can be integrated to Saleor, or the payment gateway will direct you to an external payment page. The payment gateway sends information if the payment is successful, along with tokenized credit card payment information. This token is then used to run the [`checkoutPaymentCreate`](developer/api-reference/mutations/checkout-payment-create.mdx) mutation.
```
[the remark this answers is not in the document]
Comment (korycins) [reply docs/developer/checkout.mdx:775]: It refers to the old payment flow, where the result of some payment action on the storefront was passed to Saleor core.

File: docs/developer/checkout.mdx (hunk 6 above)
Comment (korycins) [reply docs/developer/checkout.mdx]: This is a standard form that we're adding to all `Feature Preview`s sections. I can change it and remove `not complete` if you want

Comment (timuric) [reply docs/developer/checkout.mdx]: thanks!

File: docs/developer/checkout.mdx (hunk 5 above)
Comment (korycins) [reply docs/developer/checkout.mdx:774]: changed

**Pull request merged**